src/Core/AdminConsole/Services/Implementations/OrganizationService.cs

```diff
@@ -1800,16 +1800,19 @@ public async Task ImportAsync(Guid organizationId,
 
         // Users
 
+        var events = new List<(OrganizationUser ou, EventType e, DateTime? d)>();
+
         // Remove Users
         if (removeUserExternalIds?.Any() ?? false)
         {
-            var removeUsersSet = new HashSet<string>(removeUserExternalIds);
             var existingUsersDict = existingExternalUsers.ToDictionary(u => u.ExternalId);
-
-            await _organizationUserRepository.DeleteManyAsync(removeUsersSet
+            var removeUsersSet = new HashSet<string>(removeUserExternalIds)
                 .Except(newUsersSet)
                 .Where(u => existingUsersDict.ContainsKey(u) && existingUsersDict[u].Type != OrganizationUserType.Owner)
-                .Select(u => existingUsersDict[u].Id));
+                .Select(u => existingUsersDict[u]);
+
+            await _organizationUserRepository.DeleteManyAsync(removeUsersSet.Select(u => u.Id));
+            events.AddRange(removeUsersSet.Select(u => (new OrganizationUser { Id = u.Id, UserId = u.UserId, OrganizationId = u.OrganizationId }, EventType.OrganizationUser_Removed, (DateTime?)DateTime.UtcNow)));
```

Comment:
We should pass in the real, complete `OrganizationUser` object here. It looks like you're constructing an incomplete object based on the properties that `EventService` currently uses, but that could change in the future without this calling location knowing about it.

We should get the full `OrganizationUser` object from the database - although we already have `OrganizationUserUserDetails` in scope, so that is a bit duplicative. Perhaps instead we could change the `IEventService.LogOrganizationUserEventAsync` interface to accept a interface rather than the concrete `OrganizationUser` object, then you can use the `OrganizationUserUserDetails` you already have without writing mapping code.

Reply:
Also, on a _much_ less substantive note: breaking this line into multiple lines would help readability.

Reply:
Implemented as suggested on https://github.com/bitwarden/server/pull/4632/commits/aa362e441cf72d05fd20dedbff46820a31f3db61

Reply:
I really like how this turned out!